x-pack/legacy/plugins/apm/server/lib/index_pattern/get_dynamic_index_pattern.ts

```diff
@@ -39,9 +39,7 @@ export const getDynamicIndexPattern = async ({
 
   const indexPatternsFetcher = new IndexPatternsFetcher(
     (...rest: Parameters<APICaller>) =>
-      request.server.plugins.elasticsearch
-        .getCluster('data')
-        .callWithRequest(request, ...rest)
+      context.core.elasticsearch.adminClient.callAsCurrentUser(...rest)
```

Comment:
`callAsCurrentUser`. Wow, that name actually makes sense! :D

Reply:
Yeah, and I think the distinction between `adminClient` and `dataClient` will disappear as well: https://github.com/elastic/kibana/issues/49870

Reply:
I heard that. really nice this is being simplified!